sky/dashboard/src/data/connectors/jobs.jsx

```diff
@@ -189,28 +192,50 @@ export async function streamManagedJobLogs({
   signal,
   onNewLog,
 }) {
-  const timeout = 10000; // Default timeout of 10 seconds
+  // Use intelligent timeout based on activity, not arbitrary time limit
+  const inactivityTimeout = 30000; // 30 seconds of no data activity
+  let lastActivity = Date.now();
+  let timeoutId;
+
+  // Create an activity-based timeout promise
+  const createTimeoutPromise = () => {
+    return new Promise((resolve) => {
+      const checkActivity = () => {
+        const timeSinceLastActivity = Date.now() - lastActivity;
 
-  // Create a timeout promise that resolves after the specified time
-  const timeoutPromise = new Promise((resolve) => {
-    setTimeout(() => {
-      resolve({ timeout: true });
-    }, timeout);
-  });
+        if (timeSinceLastActivity >= inactivityTimeout) {
+          resolve({ timeout: true });
+        } else {
+          // Check again after remaining time
+          timeoutId = setTimeout(
+            checkActivity,
+            inactivityTimeout - timeSinceLastActivity
+          );
+        }
+      };
+
+      timeoutId = setTimeout(checkActivity, inactivityTimeout);
+    });
+  };
+
+  const timeoutPromise = createTimeoutPromise();
```

Comment:
Seems like we can just inline this function since we define it then immediately call it.

Reply:
Actually it's whatever, probably not worth it to try and fix.